ui/pages/confirmations/components/confirm/info/contexts/advanced-details-context.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {
+  ReactElement,
+  createContext,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import { useSelector } from 'react-redux';
+import { getSendHexDataFeatureFlagState } from '../../../../../../ducks/metamask/metamask';
+import { getUseNonceField } from '../../../../../../selectors';
+
+type AdvancedDetailsHandlerContextType = {
+  showAdvancedDetails: boolean;
+  setShowAdvancedDetails: (value: boolean) => void;
+};
+
+export const AdvancedDetailsHandlerContext = createContext<
+  AdvancedDetailsHandlerContextType | undefined
+>(undefined);
+
+export const AdvancedDetailsProvider: React.FC<{
```

Comment:
I think we better to add couple unit tests of this provider, since the main logic of this context provided there. 
Something like
```
it('initializes showAdvancedDetails based on Redux state', () => {
    getSendHexDataFeatureFlagStateMock.mockReturnValue(false);
    getUseNonceFieldMock.mockReturnValue(true);

    // Render the provider and a test consumer
    const { result } = renderHook(() => useAdvancedDetailsHandler(), {
      wrapper: AdvancedDetailsProvider,
    });

    // Assert that showAdvancedDetails is initialized correctly
    expect(result.current.showAdvancedDetails).toBe(true);
  });
  ```